JPetMC.cpp

```diff
@@ -21,20 +21,30 @@
 #include <G4UIExecutive.hh>
 #include <G4INCLRandom.hh>
 #include <G4UImanager.hh>
-#include <Randomize.hh>
-#include <time.h>
+#include <chrono>
+#include <thread>
+#include "Info/EventMessenger.h"
+#include <fstream> 
 
-int main (int argc, char** argv)
+void setRandomSeed()
 {
-  CLHEP::HepRandom::setTheEngine(new CLHEP::RanecuEngine());
-  G4long seed = time(NULL);
-  CLHEP::HepRandom::setTheSeed(seed);
   long seeds[2];
-  time_t systime = time(NULL);
-  seeds[0] = (long) systime;
-  seeds[1] = (long) (systime * G4UniformRand());
+  seeds[0] =  std::hash<std::thread::id>()(std::this_thread::get_id());
+  seeds[1] = (long) static_cast<long unsigned int>(std::chrono::high_resolution_clock::now().time_since_epoch().count());
```

Comment:
In case of current version of the code this is not taken into account. We are using default random generator engine which is "HepJamesRandom" which in the documentation about setSeeds() says:
"Sets the state of the algorithm according to the zero terminated array of seeds. Only the first seed is used"

Reply:
Ok

Reply:
Thanks for pointing this out